pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1049,6 +1069,39 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 		}
 	}
 
+	// TODO where to store harcoded strings
+	if env.Mesh.GetEnableAutoMtls().Value {
+		// Per envoy docs
+		// If an endpoint metadata's value under *envoy.transport_socket* does not match any
+		// *TransportSocketMatch*, socket configuration fallbacks to use the *tls_context* or
+		// *transport_socket* specified in this cluster
+		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
+			{
+				Name: "enableMtls",
+				Match: &structpb.Struct{
+					Fields: map[string]*structpb.Value{
+						util.AcceptMTLSMetadataLabel: {Kind: &structpb.Value_StringValue{StringValue: "true"}},
+					},
+				},
+				TransportSocket: &core.TransportSocket{
+					Name: "tls",
+					ConfigType: &core.TransportSocket_Config{
+						Config: &structpb.Struct{
+							Fields: map[string]*structpb.Value{},
```

Comment:
do we even need the Config here? cc @lizan

Reply:
We do need them. And even more, all the helm option that affects the upstream TLS context has to be carried out here...
Here's a working example, https://github.com/envoyproxy/envoy/pull/8354/commits/3de2b7b44f84f22da8979a8e31db5ca2b40a8c7d#diff-9a5b84a2b6b296fc9f2c8669b31181e8L46

Reply:
The `enableMTLS` should be equal to the value set in the `applyUpstreamTLSSettings`
And we need one more `enablePlaintext` equal to the `raw_buffer` transport socket to fall back to plain text for non `mTLSReady` endpoint.